cs/docs/integrace-do-aplikace.md

```diff
@@ -0,0 +1,114 @@
+---
+title: Integrace Lean Mapperu do aplikace
+---
+
+* [Úvod](#page-title)
+* [Vytvoření připojení](#toc-pripojeni)
+* [Mapper](#toc-mapper)
+* [EntityFactory](#toc-entityfactory)
+* [Repositáře](#toc-repositare)
+* [Integrace do Nette](#toc-nette)
+    * [Předání repositáře do presenteru](#toc-nette-presenter)
+
+
+## Vytvoření připojení  {#toc-pripojeni}
+
+Vytvoříme si objekt `LeanMapper\Connection` a předáme mu parametry potřebné pro připojení k databázi.
+
+``` php?start_inline=1
+$connection = new LeanMapper\Connection(array(
+    'driver'   => 'mysqli',
+    'host'     => 'localhost',
+    'username' => 'root',
+    'password' => '***',
+    'database' => 'mydatabase',
+));
+```
+
+**Tip:** `LeanMapper\Connection` přebírá stejné parametry jako třída [`DibiConnection`](https://api.dibiphp.com/3.0/Dibi.Connection.html).
+
+
+## Mapper  {#toc-mapper}
+
+Dále potřebujeme objekt implementující [`LeanMapper\IMapper`](/cs/docs/mapper/). Lean Mapper nám poskytuje výchozí implementaci formou třídy `LeanMapper\DefaultMapper`.
+
+``` php?start_inline=1
+$mapper = new LeanMapper\DefaultMapper;
+```
+
+
+## Entity factory  {#toc-entityfactory}
+
+A jako poslední věc vytvoříme objekt, který implementuje rozhraní [`LeanMapper\IEntityFactory`](/cs/docs/entity-factory). Lean Mapper nám opět podává pomocnou ruku prostřednictvím třídy `LeanMapper\DefaultEntityFactory`.
+
+``` php?start_inline=1
+$entityFactory = new LeanMapper\DefaultEntityFactory;
+```
+
+
+## Repositáře  {#toc-repositare}
+
+Abychom mohli pracovat s [entitami](/cs/docs/entity/), potřebujeme k tomu [repositáře](/cs/docs/repositare/). Předpokládejme, že máme repositář napsaný, nyní ho jen vytvoříme a předáme mu potřebné závislosti.
+
+``` php?start_inline=1
+$bookRepository = new Model\BookRepository($connection, $mapper, $entityFactory);
+```
+
+A to je vše.
+
+
+## Integrace do Nette aplikace  {#toc-nette}
+
+Ještě si ukážeme, jak použít Lean Mapper v rámci aplikace napsané v [Nette](https://nette.org). Do konfiguračního souboru aplikace (`config.neon`) si přídáme následující parametry a definice služeb:
+
+``` yaml
+parameters:
+    # údaje pro připojení k DB
+    leanmapper:
+        driver: mysqli
+        host: localhost
+        username: ...
+        password: ...
+        database: mydatabase
+
+# ...
+
+services:
+    # registrace Lean Mapperu
+    - LeanMapper\Connection(%leanmapper%)
+    - LeanMapper\DefaultMapper
+    - LeanMapper\DefaultEntityFactory
```

Comment:
Rozšíření pro Nette nepoužívám, ale ano, i to je jedna z cest. Ideální by asi bylo zmínit oboje - DI rozšíření jako doporučenou cestu a tohle nastavení jako alternativu pro ty, kteří nechtějí nemůžou DI rozšíření použít.